scripts/node/describe-nodes/script.sh

```diff
@@ -53,120 +29,43 @@ function list-selector() {
-      ;;
-    -m|--master)
-      master=true
-      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
-      ;;
-    -i|--infra)
-      infra=true
-      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
-      ;;
-    -w|--worker)
-      worker=true
-      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
-      ;;
-    -d|--debug)
-      set -x  # Print expanded commands with arguments
-      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
-      ;;
-    -n|--node|--nodes)
-      if [ -n "${nodes}" ]; then
-        error-out "Only 1 '--nodes' argument can be provided"
-      fi
-      # Check for unset or empty arguments and values that look like a flag/switch
-      if [ ${#ARG_ARRAY[@]} -le 1 ] || [ -z "${ARG_ARRAY[1]}" ] || [[ "${ARG_ARRAY[1]}" =~ ^-.* ]]; then
-        error-out "The '--nodes' argument requires a list of nodes to be provided"
-      fi
-      # Check for illegal characters in the list of nodes
-      badChars="$(echo "${ARG_ARRAY[1]}" | tr -d '[:alnum:]' | tr -d '-' | tr -d '.' | tr -d ',')"
-      if [ -n "${badChars}" ]; then
-        error-out "The '--nodes' argument can only contain '.|-|,|<alphaNumeric>' characters.  These characters '${badChars}' are illegal"
-      fi
-      nodes="${ARG_ARRAY[1]}"
-      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
-      ;;
-    -l|--selector)
-      if [ -n "${selector}" ]; then
-        error-out "Only 1 '--selector' argument can be provided"
-      fi
-      # Check for unset or empty arguments and values that look like a flag/switch
-      if [ ${#ARG_ARRAY[@]} -le 1 ] || [ -z "${ARG_ARRAY[1]}" ] || [[ "${ARG_ARRAY[1]}" =~ ^-.* ]]; then
-        error-out "The '--selector' argument must contain a node selector"
-      fi
-      badChars="$(echo "${ARG_ARRAY[1]}" | tr -d '[:alnum:]' | tr -d '-' | tr -d '.'  | tr -d '/' | tr -d ',' | tr -d ':' | tr -d ' ' | tr -d '(' | tr -d ')' | tr -d '!' | tr -d '=')"
-      if [ -n "${badChars}" ]; then
-        error-out "The '--selector' argument must not contain illegal characters.  These characters '${badChars}' are illegal"
-      fi
-      selector="${ARG_ARRAY[1]}"
-      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
-      ;;
-    *)
-      error-out "Unknown option or argument '${ARG_ARRAY[0]}'"
-      ;;
-  esac
-done
-
-# If --nodes is supplied use that and ignore all other arguments
-# else if --selector is supplied use that ignore the others arguments
-# else if --all is supplied use that ignore the others arguments
-# else if any of the --master or --infra or --worker arguments are supplied use them all
-if [ -z "${nodes}" ]
+    add2nodes "${NODES}"
+elif [ -n "${SELECTOR-}" ]
+then
+    add2nodes "$(list-selector "${SELECTOR}")"
+elif [ "${ALL:-false}" = "true" ]
 then
-  if [ -n "${selector}" ] # retest
-  then
-    nodes="$(list-selector "${selector}")"
-  elif [ "${all}" = true ]
-  then
     add2nodes "$(list-selector "node-role.kubernetes.io/master")"
```

Comment:
A nitpick comment - why add the nodes by 3 separate label selectors instead of oc get nodes?